### PR TITLE
Added user id validation

### PR DESCRIPTION
**Pull request by Anton-Shutik**
Added member id validation. That will prevent from getting unexpected data to the redis and when querying users by that ids in future.

### CODE REVIEW

File: segments/helpers.py [hunk 1]
```diff
@@ -165,21 +165,31 @@ def run_pipeline(self, iterable, operation=lambda pipeline, user_id: None):
...
+
+        logger.info(f'SEGMENTS: {value} is not valid member id')
+        return False
+
+    def execute_raw_user_query(self, sql):
+        """
+        Helper that returns an array containing a RawQuerySet of user ids and their total count.
+        """
+        if sql is None or not type(sql) == str or 'select' not in sql.lower():
+            return
```
Comment (sramirezgroveco) [inline segments/helpers.py]: Should this throw an exception? This is not a valid query to run, right?

File: segments/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    Any SQL issues encountered when Segments executing their SQL definitions will raise this exception.
+    """
+    pass
+
+
+class SegmentDefinitionUnescaped(SegmentException):
+    """
+    Raised when an unescaped percent sign is encountered
+    """
+    pass
```
Comment (Anton-Shutik) [inline segments/exceptions.py]: Just moved exceptions to separate file

File: segments/helpers.py [hunk 3]
```diff
@@ -165,21 +167,33 @@ def run_pipeline(self, iterable, operation=lambda pipeline, user_id: None):
...
+            cursor.execute(sql)
+
+            chunk = 1  # just need for 1st iteration
+            while chunk:
+                chunk = cursor.fetchmany(size=app_settings.SEGMENTS_CURSOR_FETCHMANY_SIZE)
+                for row in chunk:
+                    if self.is_valid_member_id(row[0]):
+                        yield row[0]
+                    else:
+                        logger.exception(f'Query returned invalid result: {row[0]}')
```
Comment (Anton-Shutik) [inline segments/helpers.py]: It logs an exception on invalid results, but continues to return valid results

File: segments/tests/test_helpers.py [hunk 4]
```diff
@@ -10,86 +12,88 @@
...
     @patch('segments.tasks.delete_segment.delay')
     def test_delete_segment(self, p_delete_segment):
-        s = SegmentFactory()
-        s.add_member(self.user)
-        s.delete()
-        self.assertFalse(self.helper.segment_has_member(s.id, self.user.id))
+        self.segment.add_member(self.user)
+        self.segment.delete()
+        self.assertFalse(self.helper.segment_has_member(self.segment.id, self.user.id))
         self.assertTrue(p_delete_segment.called)
```
Comment (Anton-Shutik) [inline segments/tests/test_helpers.py]: Mostly just moved `s = SegmentFactory()` to `setUp` method

File: segments/tests/test_models.py [hunk 5]
```diff
@@ -32,11 +37,6 @@ def test_user_belongs_to_segment(self):
         s.refresh()
         self.assertTrue(s.has_member(self.u))
 
-    def test_user_adds_to_segment(self):
-        s = SegmentFactory()
-        s.add_member(self.u)
-        self.assertTrue(s.has_member(self.u))
```
Comment (Anton-Shutik) [inline segments/tests/test_models.py]: Duplicate of this one https://github.com/groveco/django-segments/blob/master/segments/tests/test_helpers.py#L24

**Pull request merged**